File: tests/go/cases/case_104_indirect_container_write_01.go
package escape_tests

var retainedCase104 = []map[string]string{}

func Case104IndirectContainerWrite01(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "indirect_container_write_01",
		"entity": "stress",
		"stage": "evaluation",
		"input": raw,
		"checksum": "indirect_container_write_01:" + raw,
	}
	envelope := map[string]map[string]string{"wrapped": payload}
	// ESCAPE: indirection writes payload into retained sink.
	retainedCase104 = append(retainedCase104, envelope["wrapped"])
	return "ok"
}

File: tests/nodejs/cases/case_104_indirect_container_write_01.js
/**
 * Task case 104: indirect_container_write_01 false-positive/false-negative stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case104IndirectContainerWrite01(input) {
  const taskName = 'indirect_container_write_01';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'stress',
    stage: 'evaluation',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const envelope = { wrapped: payload };
  // ESCAPE: payload stored via indirection into retained sink.
  retainedAudit.push(envelope.wrapped);
  return 'ok';
}

module.exports = {
  case104IndirectContainerWrite01,
};
